server/src/pages/shot/editor.js

```diff
@@ -2,53 +2,168 @@ const React = require("react");
 const { Localized } = require("fluent-react/compat");
 const sendEvent = require("../../browser-send-event.js");
 
+let mousedown = false;
+let resizing = false;
+let creating = false;
+let resizeDirection;
+let resizeStartPos;
+let resizeStartSelected;
+let selectedPos = {};
+let mousedownPos = {};
+const minWidth = 10;
+const minHeight = 10;
+
+const movements = ["topLeft", "top", "topRight", "left", "right", "bottomLeft", "bottom", "bottomRight"];
+const movementPositions = {
+  topLeft: ["x1", "y1"],
+  top: [null, "y1"],
+  topRight: ["x2", "y1"],
+  left: ["x1", null],
+  right: ["x2", null],
+  bottomLeft: ["x1", "y2"],
+  bottom: [null, "y2"],
+  bottomRight: ["x2", "y2"],
+  move: ["*", "*"]
+};
+
+class Selection {
+  constructor(x1, y1, x2, y2) {
+    this.x1 = x1;
+    this.y1 = y1;
+    this.x2 = x2;
+    this.y2 = y2;
+  }
+
+  get top() {
+    return Math.min(this.y1, this.y2);
+  }
+  set top(val) {
+    if (this.y1 < this.y2) {
+      this.y1 = val;
+    } else {
+      this.y2 = val;
+    }
+  }
+
+  get bottom() {
+    return Math.max(this.y1, this.y2);
+  }
+  set bottom(val) {
+    if (this.y1 > this.y2) {
+      this.y1 = val;
+    } else {
+      this.y2 = val;
+    }
+  }
+
+  get left() {
+    return Math.min(this.x1, this.x2);
+  }
+  set left(val) {
+    if (this.x1 < this.x2) {
+      this.x1 = val;
+    } else {
+      this.x2 = val;
+    }
+  }
+
+  get right() {
+    return Math.max(this.x1, this.x2);
+  }
+  set right(val) {
+    if (this.x1 > this.x2) {
+      this.x1 = val;
+    } else {
+      this.x2 = val;
+    }
+  }
+
+  get width() {
+    return Math.abs(this.x1 - this.x2);
+  }
+  get height() {
+    return Math.abs(this.y1 - this.y2);
+  }
+
+    /** Sort x1/x2 and y1/y2 so x1<x2, y1<y2 */
+  sortCoords() {
+    if (this.x1 > this.x2) {
+      let tmp = this.x2;
+      this.x2 = this.x1;
+      this.x1 = tmp;
+    }
+    if (this.y1 > this.y2) {
+      let tmp = this.y2;
+      this.y2 = this.y1;
+      this.y1 = tmp;
+    }
+  }
```

Comment:
If you call `sortCoords()` at the end of the constructor and at the beginning of the setters, then we won't need the `Math.min()` and `Math.max()` calls on the getters, and the setters won't need their own `if`s.